src/main/scala/de/bripkens/ha/App.scala

```diff
@@ -23,7 +23,7 @@ object App extends scala.App {
   implicit val system = ActorSystem("ha")
 
   // the AppActor gets us started from here on out
-  system.actorOf(Props(classOf[AppActor], mapper, configuration), "app")
+  system.actorOf(AppActor.props(mapper, configuration), AppActor.Name)
```

Comment:
Both are saft wrt refactorings – other than the crappy `Props[Foo]` – but in the case of remoting – and sooner or later you will go remote with Akka – `Props` must be serializable. If you create an instance with the constructor as by-name argument you effectively close (like in closure) over the outer instance (thanks, JVM!) and if that's an actor – like in most cases – your `Props` aren't serializable.

So, it's a best practice / pattern to have a `Props` factory method in the companion object which in 99.9999% of the cases is serializable.